File: app/javascript/channels/game_channel.js
// // game_channel_subscription.js
// import consumer from "channels/consumer";

// function getCookie(cname) {
//   let name = cname + "=";
//   let ca = document.cookie.split(';');
//   for(let i = 0; i < ca.length; i++) {
//     let c = ca[i];
//     while (c.charAt(0) == ' ') {
//       c = c.substring(1);
//     }
//     if (c.indexOf(name) == 0) {
//       return c.substring(name.length, c.length);
//     }
//   }
//   return "";
// }

// const room_id = getCookie("room_id");

// export const gameChannelSubscription = consumer.subscriptions.create({ channel: 'GameChannel', id: room_id }, {
//   connected() {
//     console.log("Connected to GameChannel with room id :"+ room_id);
//   },
  
//   received(data) {
//       console.log("Received data:", data);
//       debugger
//   },

//   sendMove(data) {
//     console.log("send data to cable")
//     this.perform("move",  data);
//   }
// });

File: app/channels/game_channel.rb
class GameChannel < ApplicationCable::Channel
  def subscribed
    begin
      rooms = JSON.parse($redis.get("room_#{params[:id]}"))
      room_id = rooms['room_id']
      puts "stream for room_#{room_id}"
      stream_for "room_#{room_id}"
    rescue => e
      stream_for "game_channel"
      puts "Error: #{e.message}"
    end
  end

  def unsubscribed
    stop_all_streams
  end

  def move(data)
    puts "-------------------------move------------------------------------"
    puts data
  end

  def receive(data)
    puts "====================received======================================="
    puts data 
  end
end
